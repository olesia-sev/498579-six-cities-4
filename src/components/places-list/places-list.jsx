import React from 'react';
import {offersTypeArray} from '../../prop-types/prop-types';
import {PlaceCard} from "../place-card/place-card";
import PropTypes from "prop-types";
import {connect} from "react-redux";

export const MAIN_THEME = `main`;
export const NEARBY_THEME = `nearby`;
const themes = {
  [MAIN_THEME]: {
    placesList: `cities__places-list places__list tabs__content`,
  },
  [NEARBY_THEME]: {
    placesList: `near-places__list places__list`,
  },
};

const PlacesList = ({theme, offers}) => {

  const currentTheme = themes[theme];

  return (
    <div className={currentTheme.placesList}>
      {
        offers.map((offer) => (
          <PlaceCard key={offer.id} theme={theme} offer={offer} />
        ))
      }
    </div>
  );
};

PlacesList.propTypes = {
  theme: PropTypes.oneOf([MAIN_THEME, NEARBY_THEME]).isRequired,
  offers: offersTypeArray,
};

const mapStateToProps = (state) => {
  const currentCityOffers = state.offers.filter((offer) => offer.cityId === state.activeCityId);
  return {
    offers: currentCityOffers,
  };
};

export default connect(mapStateToProps)(PlacesList);
