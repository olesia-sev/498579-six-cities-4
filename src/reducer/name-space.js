const NameSpace = {
  DATA: `DATA`,
  APP: `APP`,
  USER: `USER`,
};

export {NameSpace};
