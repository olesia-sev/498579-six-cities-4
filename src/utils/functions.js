export const getRandomNumberInRange = (min, max) => {
  return Math.floor(Math.random() * (max - min + 1)) + min;
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};
