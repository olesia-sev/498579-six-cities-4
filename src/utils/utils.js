export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const SORT_TYPES = {
  POPULAR: `Popular`,
  TO_HIGH_PRICE: `Price: low to high`,
  TO_LOW_PRICE: `Price: high to low`,
  TOP_RATED: `Top rated first`,
};

export const getSortedOffers = (offers, activeSort) => {
  switch (activeSort) {
    case SORT_TYPES.POPULAR:
      return offers.slice();
    case SORT_TYPES.TO_HIGH_PRICE:
      return offers.slice().sort((a, b) => a.price - b.price);
    case SORT_TYPES.TO_LOW_PRICE:
      return offers.slice().sort((a, b) => b.price - a.price);
    case SORT_TYPES.TOP_RATED:
      return offers.slice().sort((a, b) => b.rating - a.rating);
    default:
      return offers.slice();
  }
};

export const MAX_REVIEWS_LENGTH = 10;

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  FAVOURITES: `/favorites`,
  OFFER_DETAIL: `/offers/:id`,
};

export const RATING_STAR_WIDTH = 20;

export const MAX_IMAGES_AMOUNT = 6;
export const SLICE_BEGIN = 0;
