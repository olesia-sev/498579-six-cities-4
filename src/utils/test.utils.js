import {MAIN_THEME} from "../components/places-list/places-list";
import {PLACE_CARD_FAV_BTN} from "../components/common/favourite-button/favourite-button";
import {PLACE_CARD_THEME} from "../components/common/ratinig/ratinig";

export const optionsAmount = 100;

export const testString = `Test string`;

export const emptyFunction = () => {};

export const cardId = 12;

export const theme = MAIN_THEME;
export const favButtonTheme = PLACE_CARD_FAV_BTN;
export const ratingTheme = PLACE_CARD_THEME;

export const cardDataArray = [
  {
    id: 10,
    cityId: 100,
    city: `Paris`,
    coords: [52.3909553943508, 4.85309666406198],
    img: `https://placedog.net/260/200?id=2`,
    images: [
      `https://placedog.net/260/200?id=3`,
      `https://placedog.net/260/200?id=33`,
      `https://placedog.net/260/200?id=12`,
      `https://placedog.net/260/200?id=10`,
      `https://placedog.net/260/200?id=11`,
      `https://placedog.net/260/200?id=15`,
    ],
    price: 150,
    title: `Title`,
    placeType: `Flat`,
    numberOfBedrooms: 2,
    maxGuests: 3,
    rating: 4,
    saved: true,
    premium: true,
    featuresInside: [
      `Wi-Fi`, `Heating`,
    ],
    hostName: `Pete`,
    hostAvatar: `https://placedog.net/260/200?id=5`,
    userPro: true,
    description: `Description`,
    reviews: [
      {
        id: 95,
        userName: `Mini`,
        userAvatar: `https://placedog.net/260/200?id=5`,
        rating: 4,
        content: `Review`,
        date: `April 2019`
      },
    ],
  },
  {
    id: 100,
    cityId: 200,
    city: `Cologne`,
    coords: [52.369553943508, 4.85309666406198],
    img: `https://placedog.net/260/200?id=3`,
    images: [
      `https://placedog.net/260/200?id=33`,
      `https://placedog.net/260/200?id=99`,
      `https://placedog.net/260/200?id=89`,
      `https://placedog.net/260/200?id=29`,
      `https://placedog.net/260/200?id=13`,
      `https://placedog.net/260/200?id=16`,
    ],
    price: 150,
    title: `Title test`,
    placeType: `Flat`,
    numberOfBedrooms: 1,
    maxGuests: 2,
    rating: 4,
    saved: false,
    premium: false,
    featuresInside: [
      `Wi-Fi`, `Heating`,
    ],
    hostName: `Pete`,
    hostAvatar: `https://placedog.net/260/200?id=57`,
    userPro: false,
    description: `Description`,
    reviews: [
      {
        id: 92,
        userName: `Max`,
        userAvatar: `/img/avatar-max.jpg`,
        rating: 4,
        content: `Review`,
        date: `April 2019`
      },
      {
        id: 93,
        userName: `Tom`,
        userAvatar: `/img/avatar.svg`,
        rating: 5,
        content: `Review`,
        date: `May 2019`
      },
    ],
  }
];

export const citiesArray = [
  {
    name: `Paris`,
    id: 100,
  },
  {
    name: `Cologne`,
    id: 200,
  },
  {
    name: `Brussels`,
    id: 300,
  },
  {
    name: `Amsterdam`,
    id: 400,
  },
  {
    name: `Hamburg`,
    id: 500,
  },
  {
    name: `Dusseldorf`,
    id: 600,
  }
];
