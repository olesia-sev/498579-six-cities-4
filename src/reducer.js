import {extend} from './utils/functions';
import cities from "./mocks/cities";
import offers from "./mocks/offers";

export const initialState = {
  activeCityId: 10,
  cities,
  offers,
};

const ActionType = {
  SET_ACTIVE_CITY: `SET_ACTIVE_CITY`,
};

const ActionCreator = {
  setActiveCity: (id) => ({
    type: ActionType.SET_ACTIVE_CITY,
    payload: id,
  })
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_ACTIVE_CITY:
      return extend(state, {
        activeCityId: action.payload,
      });

    default: {
      return state;
    }
  }
};

export {reducer, ActionType, ActionCreator};
