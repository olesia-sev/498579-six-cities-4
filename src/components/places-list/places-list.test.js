import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {MainPlacesList} from "./places-list";
import {cardDataArray, theme} from "../../utils/test.utils";
import {BrowserRouter as Router} from "react-router-dom";
import {NameSpace} from "../../reducer/name-space";
import {SORT_TYPES} from "../../utils/utils";

const mockStore = configureStore([]);

it(`Places list should be rendered`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      offers: cardDataArray,
    },
    [NameSpace.APP]: {
      activeSortingType: SORT_TYPES.POPULAR,
    },
  });

  const tree = renderer
    .create(<Router>
      <Provider store={store}>
        <MainPlacesList theme={theme} />
      </Provider>
    </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
