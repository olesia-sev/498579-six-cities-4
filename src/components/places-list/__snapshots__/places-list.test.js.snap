// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Places list should be rendered 1`] = `
<div
  className="cities__places-list places__list tabs__content"
/>
`;
